syntax = "proto3";

package transfer;

message TransferMetadataRequest{
    uint64 total_size = 1;
    repeated FileInfoRequest files = 2;
}

message FileInfoRequest {
    string relative_path = 1;
    uint64 size = 2;
    string hash = 3;
}

message FileInfoResponse {
    string relative_path = 1;
    enum Status {
        SUCCESS = 0;
        SKIPPED = 1;
        FAILURE = 2;
    }
    Status status = 2;
    string message = 3;
}

message TransferMetadataResponse {
    enum Status {
        SUCCESS = 0;
        FAILURE = 1;
        IN_PROGRESS = 2;
        READY = 3;
    }
    Status status = 1;
    string message = 2;
}

message FileChunkRequest {
    string file_relative_path = 1;
    uint64 chunk_index = 2;
    bytes data = 3;
    string hash = 4;
    bool is_last_chunk = 5;
}

message FileChunkResponse {
    string file_relative_path = 1;
    uint64 chunk_index = 2;
    enum Status {
        RECEIVED = 0;
        FAILURE = 1;
    }
    Status status = 3;
    string message = 4;
}